ui/index.js

```diff
@@ -41,6 +43,9 @@ export const updateBackgroundConnection = (backgroundConnection) => {
   backgroundConnection.onNotification((data) => {
     if (data.method === 'sendUpdate') {
       reduxStore.dispatch(actions.updateMetamaskState(data.params[0]));
+      browser.storage.local.set({ [STORE_KEY]: data.params[0] }).then(() => {
```

Comment:
when we get an update to state from the bg we save the snapshot to the storage. In addition to what normally happens.